sdk/appcenter-storage/src/test/java/com/microsoft/appcenter/storage/StorageTest.java

```diff
@@ -267,12 +388,43 @@ public void buildAppCenterGetDbTokenBodyPayload() {
         // These constructors must be called even though these classes are not going to be instantiated.
         TokenExchange te = new TokenExchange();
         CosmosDb cdb = new CosmosDb();
+        Utils utils = new Utils();
+        Utils.handleApiCallFailure(new HttpException(200, "b"));
+    }
+
+    @Test
+    public void documentDeserialization() {
+        Document<TestDocument> d =
+                Utils.parseDocument(cosmosDbDocumentResponsePayload, TestDocument.class);
+        assertEquals(DOCUMENT_ID, d.getId());
+        assertEquals(PARTITION, d.getPartition());
+        assertEquals(TEST_FIELD_VALUE, d.getDocument().test);
+    }
+
+    @Test
+    public void documentSerialization() {
+        assertEquals(
+                TEST_FIELD_VALUE,
+                Utils.fromJson(
+                        String.format("{\"test\": \"%s\"\n" + "}", TEST_FIELD_VALUE),
+                        TestDocument.class).test);
+    }
+
+    @Test
+    public void generateHeaders() {
+        Map<String, String> headers = CosmosDb.generateHeaders(PARTITION, "token");
+        assertEquals(5, headers.size());
+    }
+
+    @Test(expected = IllegalArgumentException.class)
+    public void urlEncogingThrowsNonExistingEncoding() {
+        CosmosDb.urlEncode("a string to encode", "An encoding that doesn't exist");
     }
 
     class TestDocument {
```

Comment:
Found it! ;)

Reply:
(Use TestDocument from top level class you defined)

Reply:
Hm, that is some funny refactoring corner case there! Because I used `Refactor` to move it to the separate file. But for some reason it didn't erase the original one. Good eye!